aten/src/ATen/native/cuda/TensorModeKernel.cuh

```diff
@@ -374,43 +383,43 @@ __global__ void compute_mode(
   }
   __syncthreads(); // broadcast mode
 
-  // Finally, we need to find the "an" index of the mode in the input Tensor.
-  // The API does not constrain which index we pick, so it can be any of the
-  // indices that contain the mode. We will do a reduction to find the index. We
-  // go back to using the (index, flag) buffer arrangement. First, we mark
-  // indices that are equal to the mode, i.e B[i] = true if input[i] == mode,
-  // and initialize C[i] to be the index
+  // Finally, we need to find "an" index of the mode in the input
+  // Tensor. The API does not constrain which index we pick, but here
+  // we always pick the largest index. We store the index if the value
+  // is the mode, or 0 otherwise. Then find the maximum value.
```

Comment:
This fixes the ROCm failures. The old code happened to always pick the largest index due to the way the reductuion was structured. However, the warp shuffle based reduction has a different order of comparisons, so didn't have the same end result. This operator actually guarantees the largest index is returned, regardless of the reduction order.

Reply:
Out of curiosity, doesn't WARP_SHFL_DOWN also always the last (largest) index in the end?

Reply:
`WarpReduce` goes from biggest offset to smallest offset. So, if priority goes to the right hand side then element 7 has higher priority than 8 or 9 because 8 = 0 mod 8 and 9 = 1 mod 8 are to the left of 7 = 7 mod 8.